store/storage/rocksdb/opts.go

```diff
@@ -0,0 +1,122 @@
+package rocksdb
+
+import (
+	"encoding/binary"
+	"runtime"
+
+	"github.com/linxGnu/grocksdb"
+)
+
+const (
+	// CFNameStateStorage defines the RocksDB column family name for versioned state
+	// storage.
+	CFNameStateStorage = "state_storage"
+
+	// CFNameDefault defines the RocksDB column family name for the default column.
+	CFNameDefault = "default"
+)
+
+// NewRocksDBOpts returns the options used for the RocksDB column family for use
+// in state storage.
+//
+// FIXME: We do not enable dict compression for SSTFileWriter, because otherwise
+// the file writer won't report correct file size.
+// Ref: https://github.com/facebook/rocksdb/issues/11146
+func NewRocksDBOpts(sstFileWriter bool) *grocksdb.Options {
+	opts := grocksdb.NewDefaultOptions()
+	opts.SetCreateIfMissing(true)
+	opts.SetComparator(CreateTSComparator())
+	opts.IncreaseParallelism(runtime.NumCPU())
+	opts.OptimizeLevelStyleCompaction(512 * 1024 * 1024)
+	opts.SetTargetFileSizeMultiplier(2)
+	opts.SetLevelCompactionDynamicLevelBytes(true)
+
+	// block based table options
+	bbto := grocksdb.NewDefaultBlockBasedTableOptions()
+
+	// 1G block cache
+	bbto.SetBlockSize(32 * 1024)
+	bbto.SetBlockCache(grocksdb.NewLRUCache(1 << 30))
```

Comment:
1 billion elements seems pretty big, if block size is 32kb wouldn't the block cache be 32TB or have I misunderstood something here?
```
>>> (1024 * 32 * (1 << 30)) / 1e12
35.184372088832
```

Reply:
This was taken from versionDB -- I didn't vet all the exact params as Cronos experiences excellent result with this.

Reply:
Well if the above is correct, yes loading the entire db into page cache will produce excellent results at the cost of memory space.

Reply:
I added a method `NewWithDB` in case a chain wants complete control over how they configure RocksDB (if they so choose to use it).